### PR TITLE
add flag in photonID parameters

### PR DESCRIPTION
**Pull request by dasphy**
Follow-up of https://github.com/HEP-FCC/k4RecCalorimeter/pull/79/ .
Added the flag in photon ID parameters, to calculate variables using log E weights instead of linear E weights.

Tagging @giovannimarchiori  @BrieucF 

Cheers, Tong

### CODE REVIEW

File: FCCee/FullSim/ALLEGRO/ALLEGRO_o1_v03/run_digi_reco.py [hunk 1]
```diff
@@ -419,6 +420,7 @@
                                                    layerFieldNames=["layer"],
                                                    thetaRecalcWeights=[ecalBarrelThetaWeights],
                                                    do_photon_shapeVar=runPhotonIDTool,
+                                                   do_widthTheta_logE_weights=logEWeightInPhotonID
```
Comment (BrieucF) [inline FCCee/FullSim/ALLEGRO/ALLEGRO_o1_v03/run_digi_reco.py:423]:
```suggestion
                                                   do_widthTheta_logE_weights=logEWeightInPhotonID,
```

File: FCCee/FullSim/ALLEGRO/ALLEGRO_o1_v03/run_digi_reco.py [hunk 2]
```diff
@@ -555,6 +557,7 @@
                                                        layerFieldNames=["layer"],
                                                        thetaRecalcWeights=[ecalBarrelThetaWeights],
                                                        do_photon_shapeVar=runPhotonIDTool,
+                                                       do_widthTheta_logE_weights=logEWeightInPhotonID
```
Comment (BrieucF) [inline FCCee/FullSim/ALLEGRO/ALLEGRO_o1_v03/run_digi_reco.py:560]:
```suggestion
                                                       do_widthTheta_logE_weights=logEWeightInPhotonID,
```

**Review by BrieucF — CHANGES_REQUESTED** [review]
Hi Tong, thanks. The changes you propose break the config (see suggestion of modifications). For the next ones, can you please test that the code runs before to open the PR?

**Comment by dasphy** [comment]
Oh yes thanks Brieuc, sorry for that. Modified.

File: FCCee/FullSim/ALLEGRO/ALLEGRO_o1_v03/run_digi_reco.py [hunk 3]
```diff
@@ -98,6 +98,7 @@
 
 # run photon ID algorithm
 runPhotonIDTool = False
+logEWeightInPhotonID = False and runPhotonIDTool
```
Comment (giovannimarchiori) [inline FCCee/FullSim/ALLEGRO/ALLEGRO_o1_v03/run_digi_reco.py:101]: Hi @dasphy ,
I don't think that the flag logEWeightInPhotonID should depend on runPhotonIDTool.
The runPhotonIDTool flag is used to decide whether we want to run the photon ID BDT and save its output score in the output ROOT file, while logEWeightInPhotonID is used to decide if we want to use logE or E weights when calculating some shape variables used to decorate the clusters, no matter whether we run or not the photon ID tool later. I would just set the flag to False

**Review by giovannimarchiori — APPROVED** [review]
Looks good now

**Pull request merged**